pychunkedgraph/graph/edits.py

```diff
@@ -210,20 +241,41 @@ def add_edges(
     for cc_indices in components:
         l2ids_ = graph_ids[cc_indices]
         new_id = cg.id_client.create_node_id(cg.get_chunk_id(l2ids_[0]))
-        cg.cache.children_cache[new_id] = np.concatenate(
-            [atomic_children_d[l2id] for l2id in l2ids_]
-        )
-        cg.cache.atomic_cx_edges_cache[new_id] = concatenate_cross_edge_dicts(
-            [atomic_cross_edges_d[l2id] for l2id in l2ids_]
-        )
-        cache_utils.update(
-            cg.cache.parents_cache, cg.cache.children_cache[new_id], new_id
-        )
         new_l2_ids.append(new_id)
         new_old_id_d[new_id].update(l2ids_)
         for id_ in l2ids_:
             old_new_id_d[id_].add(new_id)
 
+        # update cache
+        # map parent to new merged children and vice versa
+        merged_children = np.concatenate([atomic_children_d[l2id] for l2id in l2ids_])
+        cg.cache.children_cache[new_id] = merged_children
+        cache_utils.update(cg.cache.parents_cache, merged_children, new_id)
+
+    # update cross chunk edges by replacing old_ids with new
+    # this can be done only after all new IDs have been created
+    updated_entries = []
+    for new_id, cc_indices in zip(new_l2_ids, components):
+        l2ids_ = graph_ids[cc_indices]
+        new_cx_edges_d = {}
+        cx_edges = [cross_edges_d[l2id] for l2id in l2ids_]
+        cx_edges_d = concatenate_cross_edge_dicts(cx_edges, unique=True)
+        temp_map = {k: next(iter(v)) for k, v in old_new_id_d.items()}
+        for layer, edges in cx_edges_d.items():
+            edges = fastremap.remap(edges, temp_map, preserve_missing_labels=True)
+            new_cx_edges_d[layer] = edges
+            assert np.all(edges[:, 0] == new_id)
+        cg.cache.cross_chunk_edges_cache[new_id] = new_cx_edges_d
+        entries = _update_neighbor_cross_edges(
```

Comment:
I think this here can introduce problems if a neighboring node is a neighbor to multiple `new_l2_ids`. 

`_update_neighbor_cross_edges` looks right to me. It writes a complete new set of L2 edges for a node. But if the same node is updated multiple updates, then only the last one is reflected. Maybe the logic here takes care of this somehow but then it still introduces multiple unnecessary writes.

So, _if_ I am correct about this, the solution would be to consolidate this call across all new_l2_ids to only make one call per neighboring node id.